assets/js/filter.js

```diff
@@ -0,0 +1,196 @@
+            var filterRow = $('<tr class="tableexport-ignore"></tr>');
+            for (var i = 0; i < columnCount; i++) {
+                this.filters.push('');
+                var filterInput = $('<input type="text" data-column="' + i + '">')
+                    .on('keyup', filterDelay).on('change', filterDelay);
+
+
+                var filterCell = $(
+                    '<td data-toggle="tooltip" title="'
+                    + _(
+                        "Comparison operators >, <, >=, <=, !=, == are available — e.g. >=200. "
+                        + "Use dot as decimal seperator — e.g. 2.1. "
+                        + "Regex is also available — e.g. /^\\d+$/g"
+                        )
+                    + '"></td>'
+                );
+                filterCell.append(filterInput);
+                filterRow.append(filterCell);
+            }
+            this.element.find('thead').append(filterRow);
+        },
+
+
+        filterGetType: function(query) {
+            if (!query) {
+                // Empty query, no filtering
+                return [undefined, ""];
+            }
+
+            var querySliceOne = query.slice(1);
+            var querySliceTwo = query.slice(2);
+            querySliceOne = isNaN(querySliceOne) ? querySliceOne : parseFloat(querySliceOne);
+            querySliceTwo = isNaN(querySliceTwo) ? querySliceTwo : parseFloat(querySliceTwo);
+
+            if (query.startsWith("<")) {
+                if (query.startsWith("<=")) {
+                    return [this.lteFilter, querySliceTwo];
+                } else {
+                    return [this.ltFilter, querySliceOne];
+                }
+
+            } else if (query.startsWith(">")) {
+                if (query.startsWith(">=")) {
+                    return [this.gteFilter, querySliceTwo];
+                } else {
+                    return [this.gtFilter, querySliceOne];
+                }
+
+            } else if (query.startsWith("==")) {
+                return [this.eFilter, querySliceTwo];
+
+            } else if (query.startsWith("!=")) {
+                return [this.neFilter, querySliceTwo];
+
+            } else if (query.match(/\//g) && query.match(/\//g).length >= 2) {
+                var parts = query.split('/');
+                var regex = query;
+                var options = "";
+
+                if (parts.length > 1) {
+                    regex = parts[1];
+                    options = parts[2];
+                }
+                try {
+                    // If no error is thrown here, the query is valid regex
+                    return [this.regexFilter, new RegExp(regex, options)];
+                } catch (e) {
+                    // If the query is not valid regex, use normal filter
+                    return [this.normalFilter, query];
+                }
+
+            } else {
+                return [this.normalFilter, query];
+            }
+
+        },
+
+        lteFilter: function(text, query) {
```

Comment:
You can move this out of the class, but keep them inside the anonymous function scope. E.g. put these below the "class".